### ISSUE TITLE
[Feature Request] Option to Always Show Progress Bar

### ISSUE THREAD

**Issue by Swishy16** [TeamNewPipe/NewPipe#7206]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### What feature do you want?

I would like the option to "always display" the progress bar so that the progress bar is always visible on the video throughout playback.  Currently, the progress bar disappears after 2-3 seconds.  This feature would be a ui option that can be turned on or off based on personal preference by each user so that once enabled or disabled it will repeat the same behavior automatically for every video.

#### Why do you want this feature?

I would like this feature so that the progress bar remains on screen as a visual representation of the time remaining in each video without having to continually tap the screen to see the progress bar and the amount of time remaining in the video.

**Comment by litetex** [comment]
I completely disagree with this feature request.
Literally every video player I know (YT browers and app player, native browser/firefox video player, windows media player, etc) hide the status bar after some time.

**Comment by TippyLion28** [comment]
> I completely disagree with this feature request. Literally every video player I know (YT browers and app player, native browser/firefox video player, windows media player, etc) hide the status bar after some time.

VLC

**Comment by litetex** [comment]
@TippyLion28 
Not on Android 😉

**Comment by TippyLion28** [comment]
Touché

**Comment by Swishy16** [comment]
> @TippyLion28 Not on Android 😉

@TippyLion28 and @litetex the "Not on Android" response is not accurate.  The Youtube app on Android indeed permits viewing the progress bar continuously on all videos by tapping the screen to see the player controls.  The player controls and progress bar will overlay on the screen and will remain on top of the video throughout the video playback and will remain visible as the as the app plays consecutive videos on a playlist.  I am not sure if NewPipe will permit this request but the idea is to have the progress bar visible without the having the player controls visible.

This request was inspired by the Android Youtube app behavior as NewPipe currently does not have this capability.

In the original Youtube web version the player progress bar was visible at all times as well.  There are various scripts available that permit users to recreate this behavior even after the recent Youtube web player UI changes.

I am not clear on why you disagree with this feature request as the request specifically states in the first sentence that this be an "option".  If you prefer to have the progress bar not visible you can simply not enable it.  The way I see it, that's a win for you and a win for me.

Hope this post clarifies my reasons for this request and addresses your disagreement @litetex.

Have a good one.

**Comment by Swishy16** [comment]
> @SameenAhnaf I assume @Swishy16 is after a solution for fullscreen mode?

@TippyLion28 Yes, that's what I'm after.

**Comment by SameenAhnaf** [comment]
These options might be added. Note that these options don't need to be shown when the Interface is shown. Because, Newpipe's player Interface doesn't obstruct progress bar or notification bar.

![IMG_20211012_094558](https://user-images.githubusercontent.com/79245641/136891161-31fb8f59-119b-47aa-81b0-b9328cbdbbd7.jpg)

<details><summary> My reply </summary>

>Please let me have this modest setting request as an option for NewPipe and leave me in peace.

I understand your reason of frustration. But this is something common in open source community. I myself had to face such backlashes multiple times. 

There are tons of Newpipe issues with dozens of dislikes and many of them are still open. @litetex never threatened to close this issue. But if I am not wrong, he's disappointed about some specific problems.
1. Showing only progress bar would make the Interface look like broken.
2. Always showing the full Interface is kinda niche as it obstructs the video playback with a dark overlay.

Anyway,‌‌ I suggested an alternative again which might suit your needs. So, please let me know if you agree with my suggestion.

>Pretty please with cherries on top. Sincerely

I like your sense of humor.

</details>

**Comment by TippyLion28** [comment]
No, it is not *already* mentioned at the top. The feature you mentioned hasn't even been added to NewPipe. If that issue goes stale and is never implemented, I imagine @Swishy16 would still like the functionality they asked for to be implemented.

If this functionality was already achieved in NewPipe I would understand your point, and I would consider an issue like this to be redundant. As the functionality does not yet exist, this issue is valid and independent of the other issues you mentioned.

**Comment by SameenAhnaf** [comment]
>No, it is not already mentioned at the top.

I was talking about the screenshot. 😑

>The feature you mentioned hasn't even been added to NewPipe.

Did you read https://github.com/TeamNewPipe/NewPipe/issues/7206#issuecomment-940646534 properly?

>>These options might be added. Note that these options don't need to be shown when the Interface is shown. Because, Newpipe's player Interface doesn't obstruct progress bar or notification bar.


>As the functionality does not yet exist, this issue is valid and independent of the other issues you mentioned.

<b> Neither @litetex nor I called any issue invalid. #5235 is not related to my new suggestion.  I am just willing to add an alternative instead. </b> As I stated in <i> My reply to @Swishy16 </i> section, adding this functionality creates two new problems. If there are alternatives to solve the problems, the alternatives should be considered.

>>1. Showing only progress bar would make the Interface look like broken.
>>
>>2. Always showing the full Interface is kinda niche as it obstructs the video playback with a dark overlay.

**Comment by Swishy16** [comment]
> > Why is the progress bar always required to be shown? Why doesn't showing only progress time get the job done? 🤔
> 
> @SameenAhnaf This I can't very well comment on. @Swishy16 is the only one who can elaborate further whether this would satisfy their requirements.
> 
> However from an accessibilty point of view, I can see this being useful for users with reduced fine motor control who may find the 2-3 second timeout to be insufficient, and the constant auto-hiding of player controls to be unhelpful. I imagine this is why YouTube classes such an feature as an accessibilty option.
> 
> I also think that the progress bar provides a better visual 'at-a-glance' view of how much of the content has already been consumed.

@TippyLion28 Agreed.  The progress bar can more easily be viewed at a glance than the elapsed time text.  Also, the elapsed time displayed as text requires interpretation of the current time elapsed in relation to the total length of the video.  Whereas the red progress bar can be seen and interpreted visually based on the length of the bar and the speed of progression across the screen

@SameenAhnaf I appreciate you providing your honest feedback and a possible solution to this post but the general idea is to have the progress bar as the visual representation of the elapsed and remaining time of the video vs seeing the actual text of the elapsed and remaining time which may not be as easily visible if the phone is not within arms length distance for viewing for those whose vision is limited, such as myself.

I just want to point out that NewPipe has many features that are not present in other apps and is marketed as an alternative to the YouTube application.  I appreciate that it's open source and that everyone has their own vision and expectations of what they would like included in the application.  I respect that everyone may not want the same features as myself and that is why I made it as clear as possible that this request be an option so that it does not force anyone to use it if their preference differs from my own.

It could be a simple setting to determine how long the progress bar remains on the screen i.e. 2 seconds, 5 seconds, never.  I don't want to mess up what I feel is a good thing for everybody.

It's just my personal preference that may or not be shared by others.  I have no idea if this request will ever be implemented or not and I didn't anticipate that making this request would get this much attention.  Just thought I would ask and hope for the best.

Have a good one.

**Comment by SameenAhnaf** [comment]
Following progress bar could be optionally shown when the Interface is hidden. No need of dark overlay!

![IMG_20211015_013233](https://user-images.githubusercontent.com/79245641/137383710-2e5a9807-ccf3-4099-88b2-2e7071b858c7.jpg)

**Comment by Swishy16** [comment]
> Following progress bar could be optionally shown when the Interface is hidden. No need of dark overlay!
> 
> ![IMG_20211015_013233](https://user-images.githubusercontent.com/79245641/137383710-2e5a9807-ccf3-4099-88b2-2e7071b858c7.jpg)

@SameenAhnaf I like that visualization as well.  What platform is the screenshot from?
